experimental/dds/ot/src/ot.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { bufferToString } from "@fluidframework/common-utils";
+import { IFluidSerializer } from "@fluidframework/core-interfaces";
+
+import {
+    FileMode,
+    ISequencedDocumentMessage,
+    ITree,
+    TreeEntry,
+} from "@fluidframework/protocol-definitions";
+import {
+    IChannelAttributes,
+    IFluidDataStoreRuntime,
+    IChannelStorageService,
+    IChannelFactory,
+    Serializable,
+} from "@fluidframework/datastore-definitions";
+import { SharedObject } from "@fluidframework/shared-object-base";
+import { Doc, type, JSONOp, replaceOp, insertOp, moveOp, removeOp } from "ot-json1";
+import { OTFactory } from "./factory";
+import { debug } from "./debug";
+import { ISharedOT, ISharedOTEvents } from "./interfaces";
+
+export class SharedOT<T extends Serializable = any>
+    extends SharedObject<ISharedOTEvents<T>>
+    implements ISharedOT<T>
+{
+    private readonly pendingOps: JSONOp[] = [];
+
+    /**
+     * Create a new shared OT
+     *
+     * @param runtime - data store runtime the new shared map belongs to
+     * @param id - optional name of the shared map
+     * @returns newly create shared map (but not attached yet)
+     */
+    public static create(runtime: IFluidDataStoreRuntime, id?: string) {
+        return runtime.createChannel(id, OTFactory.Type) as SharedOT;
+    }
+
+    /**
+     * Get a factory for SharedOT to register with the data store.
+     *
+     * @returns a factory that creates and load SharedOT
+     */
+    public static getFactory(): IChannelFactory {
+        return new OTFactory();
+    }
+
+    // eslint-disable-next-line no-null/no-null
+    private root: Doc = null;
```

Comment:
The json1 'Doc' type === Jsonable.  Plugging in support for IFluidHandle will be an interesting exercise.